File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.java
import java.util.*;
class Solution {
    public int solution(int[] arr) {
        int answer = 0;
        boolean flag = true;
        Integer[] tmp = Arrays.stream(arr).boxed().toArray(Integer[]::new);
        Arrays.sort(tmp, Comparator.reverseOrder()); // 내림차순
        int i = 0;
        while(flag){
            i++;
            int temp = tmp[0]*i;
            for(int l : tmp){
                if(temp%l!=0) {
                    flag = true;
                    break;
                }
                else flag = false;
            }
        }
        return tmp[0]*i;
    }
}

File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py
def solution(arr):
    answer = 0
    arr.sort(reverse=True)
    k = 1
    while True:
        temp = arr[0] * k
        flag = True
        for i in arr:
            if temp%i != 0:
                flag = False
                break
        if flag:
            return arr[0]*k
        k += 1
